openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp

```diff
@@ -502,4 +505,24 @@ void TrafficMode::HandlePoint(m2::PointD clickPoint, Qt::MouseButton const butto
     return;
   }
 }
+
+bool TrafficMode::StartBuildingPathChecks() const
+{
+  CHECK(m_currentSegment, ("A segment should be selected before path building is started."));
+
+  if (m_buildingPath)
```

Comment:
Could you please explain why do you mix CHECKs with exceptions? What's the reason to use exceptions here?

Reply:
Exception may be handled by ui. The situation that leads to this exception is triggered by a user.
While CHECK(m_currentSegment, (...)) is for catching logical error.

Reply:
Okay.